src/imars3d/backend/diagnostics/tilt.py

```diff
@@ -103,6 +103,7 @@ def calculate_dissimilarity(
     tilt: float,
     image0: np.ndarray,
     image1: np.ndarray,
+    center: Optional[Tuple[Union[float, int], Union[float, int]]] = None,
```

Comment:
This would be more readable if you created a [custom type](https://docs.python.org/3/library/typing.html#user-defined-generic-types). Maybe
```python
type Number = Union[float, int]
type Pair = Tuple[Number, Number]
...
center: Optional[Pair] = None,
```

Reply:
it will be easier once we make 3.11 the minimum version of python as we can write it as `Optional[Tuple[float|int, float|int]]`.

Reply:
I think that the compound typing you have is harder to read. You can keep it how you want, but `center: Optional[Tuple[Number, Number]] = None` is significantly clearer to me.